packages/orchestrator-ui-components/src/components/WFOSteps/WFOStepList/WFOStepList.tsx

```diff
@@ -104,6 +131,32 @@ export const WFOStepList = ({ steps = [], startedAt }: WFOStepListProps) => {
             </EuiFlexGroup>
             <>
                 {steps.map((step, index) => {
+                    let prevState = {};
```

Comment:
For me personally it is very hard to follow what is going on in this mapper, especially with the reassignments of prevState and delta. Try to move this logic as much as possible into a separate function, perferably in a separate ts file with unit tests.